packages/streamline/src/components/dialog/dialog.types.tsx

```diff
@@ -0,0 +1,16 @@
+import { IconsName } from '../../primitives/icon/icon.types';
+import { ButtonProps } from '../buttons/button/button.types';
+
+export interface DialogProps {
+  title: string;
+  icon?: IconsName;
+  buttons: {
+    orientation: 'horizontal' | 'vertical';
+    primary: ButtonProps & {
+      appearance: 'primary' | 'danger' | 'warning';
+    };
+    secondary?: ButtonProps & {
+      appearance: 'neutral' | 'secondary';
```

Comment:
@hcourthias @ianrobertlambert Better safe than sorry, make sure you're aligned on possible appearance for each button.